api/PayrollProcessor.Functions.Api/Features/Departments/DepartmentEmployeeTrigger.cs

```diff
@@ -60,37 +62,48 @@ public class DepartmentEmployeeTrigger
                     return;
 
                 default:
+                    log.LogWarning("Queue message with {eventName} cannot be handled", eventName);
+
                     return;
             }
         }
 
-        private Task HandleEmployeeCreation(CloudQueueMessage queueMessage, ILogger log)
-        {
-            var message = QueueMessageHandler.FromQueueMessage<EmployeeCreation>(queueMessage);
-
-            return queryDispatcher.Dispatch(new EmployeeQuery(message.EmployeeId))
-                .Bind(employee => commandDispatcher.Dispatch(new DepartmentEmployeeCreateCommand(employee, idGenerator.Generate())))
-                .Bind(departmentEmployee => apiClient.SendNotification(nameof(EmployeeUpdatesQueue), departmentEmployee))
-                .Match(
-                    _ => log.LogInformation(""),
-                    () => log.LogError(""),
-                    e => log.LogError(e, ""));
-        }
+        private Task HandleEmployeeCreation(CloudQueueMessage queueMessage, ILogger log) =>
+            queueMessage
+                .Apply(QueueMessageHandler.FromQueueMessage<EmployeeCreation>)
+                .Apply(message => queryDispatcher.Dispatch(new EmployeeQuery(message.EmployeeId))
+                    .DoIfNoneOrFail(
+                        () => log.LogError("Could not find Employee {employeeId}", message.EmployeeId),
+                        ex => log.LogError(ex, "Could not query for Employee {employeeId}", message.EmployeeId))
+                    .Bind(
+                        employee => commandDispatcher.Dispatch(new DepartmentEmployeeCreateCommand(employee, idGenerator.Generate()))
+                            .DoIfFail(ex => log.LogError(ex, "Could not create Department Employee for {@employee}", employee))
+                            .Do(de => log.LogInformation("{@departmentEmployee} created for {employeeId}", de, message.EmployeeId)))
+                    .Bind(departmentEmployee =>
+                        apiClient.SendNotification(nameof(EmployeeUpdatesQueue), departmentEmployee)
+                            .DoIfFail(ex => log.LogError(ex, "Could not send API notification for {@departmentEmployee} creation", departmentEmployee))
+                            .Do(_ => log.LogInformation($"API notification sent")))
+                    .Try());
 
-        private Task HandleEmployeeUpdate(CloudQueueMessage queueMessage, ILogger log)
-        {
-            var message = QueueMessageHandler.FromQueueMessage<EmployeeUpdate>(queueMessage);
+        private Task HandleEmployeeUpdate(CloudQueueMessage queueMessage, ILogger log) =>
+            queueMessage
+                .Apply(QueueMessageHandler.FromQueueMessage<EmployeeUpdate>)
+                .Apply(message => queryDispatcher.Dispatch(new EmployeeQuery(message.EmployeeId))
```

Comment:
We could reduce the nesting of the 2nd `.Apply()` by defining another private method with the signature of the `Func<>` passed to `.Apply()` here - `Func<EmployeeUpdate, Task<Unit>>` - and then passing that private method to `.Apply()`.